src/core/objects/object_mover.cpp

```diff
@@ -124,19 +156,35 @@ void ObjectMover::move(qint32 dx, qint32 dy, bool move_opposite_handles)
 	}
 	
 	// Apply handle constraints
-	if (move_opposite_handles)
+	for (auto& constraint : handle_constraints)
 	{
-		for (auto& constraint : handle_constraints)
+		MapCoord anchor_point = constraint.object->getCoordinate(constraint.curve_anchor_index);
+		MapCoord moved_handle = constraint.object->getCoordinate(constraint.moved_handle_index);
+		MapCoordF to_hover_point = MapCoordF(moved_handle - anchor_point);
+		to_hover_point.normalize();
+		
+		MapCoord control = constraint.object->getCoordinate(constraint.opposite_handle_index);
+
+		// Check conditions for "corner point click-in" operation
+		if (move_opposite_handles == HandleOpMode::Click
+			&& constraint.anchor_is_corner)
 		{
-			MapCoord anchor_point = constraint.object->getCoordinate(constraint.curve_anchor_index);
-			MapCoordF to_hover_point = MapCoordF(constraint.object->getCoordinate(constraint.moved_handle_index) - anchor_point);
-			to_hover_point.normalize();
-			
-			MapCoord control = constraint.object->getCoordinate(constraint.opposite_handle_index);
-			control.setX(anchor_point.x() - constraint.opposite_handle_dist * to_hover_point.x());
-			control.setY(anchor_point.y() - constraint.opposite_handle_dist * to_hover_point.y());
-			constraint.object->setCoordinate(constraint.opposite_handle_index, control);
+			if (forms_a_corner(anchor_point, moved_handle, control, corner_tolerance))
+				break; // still is a corner point -> refrain from moving opposite handle
+			else
```

Comment:
`else` after `break` is redundant.

Reply:
Ack.

Reply:
This part has been reimplemented as a `switch`.